client/allocrunner/taskrunner/task_runner_test.go

```diff
@@ -2866,6 +2866,75 @@ func TestTaskRunner_BaseLabels(t *testing.T) {
 	require.Equal(alloc.Namespace, labels["namespace"])
 }
 
+// TestTaskRunner_BaseLabels_IncludesMetadata tests that the base labels include
+// the metadata fields
+func TestTaskRunner_BaseLabels_IncludesMetadata(t *testing.T) {
+	ci.Parallel(t)
+	require := require.New(t)
+
+	alloc := mock.BatchAlloc()
+	alloc.Namespace = "not-default"
+	job := alloc.Job
+	job.Meta = map[string]string{"owner": "HashiCorp", "my-key": "my-value", "some_dynamic_value": "now()"}
+	task := job.TaskGroups[0].Tasks[0]
+	task.Driver = "raw_exec"
+	task.Config = map[string]interface{}{
+		"command": "whoami",
+	}
+
+	config, cleanup := testTaskRunnerConfig(t, alloc, task.Name, nil)
+	defer cleanup()
+
+	config.ClientConfig.IncludeMetadataInMetrics = true
+
+	tr, err := NewTaskRunner(config)
+	require.NoError(err)
+
+	labels := map[string]string{}
+	for _, e := range tr.baseLabels {
+		labels[e.Name] = e.Value
+	}
+
+	require.Equal(task.Meta["owner"], labels["owner"])
+	require.Equal(task.Meta["some_dynamic_value"], labels["some_dynamic_value"])
+	require.Equal(task.Meta["my-key"], labels["my_key"])
+}
+
+// TestTaskRunner_BaseLabels_IncludesMetadata tests that the base labels include
+// the metadata fields using the provided allowed list of keys
+func TestTaskRunner_BaseLabels_IncludesMetadata_WithAllowList(t *testing.T) {
```

Comment:
I appreciate you being fine-grained with tests, but we could probably just squash these two tests together and save some test overhead. They aren't exercising anything all that independent.